### PR TITLE
Enhance find functionality

### PR DESCRIPTION
**Pull request by fymbc**

### CODE REVIEW

File: src/main/java/seedu/address/commons/util/StringUtil.java [hunk 1]
```diff
@@ -31,7 +31,7 @@ public static boolean containsIgnoreCase(String sentence, String word) {
 
         String preppedWord = word.trim();
         checkArgument(!preppedWord.isEmpty(), "String parameter cannot be empty");
-        checkArgument(preppedWord.split("\\s+").length == 1, "String parameter should be a single word");
+        //checkArgument(preppedWord.split("\\s+").length == 1, "String parameter should be a single word");
```
Comment (AlphaPyke) [inline src/main/java/seedu/address/commons/util/StringUtil.java:34]: perhaps commented-out line could be removed for code cleanliness?

File: src/test/java/seedu/address/model/person/PersonContainsKeywordsPredicateTest.java [hunk 2]
```diff
@@ -16,115 +14,141 @@ public class PersonContainsKeywordsPredicateTest {
...
+
+        // Partial match with email
+        searchCriteria = Map.of("email", "ali");
+        predicate = new PersonContainsKeywordsPredicate(searchCriteria);
+        assertTrue(predicate.test(new PersonBuilder().withEmail("alice@example.com").build()));
+
+        // Partial match with phone
+        searchCriteria = Map.of("phone", "123");
+        predicate = new PersonContainsKeywordsPredicate(searchCriteria);
+        assertTrue(predicate.test(new PersonBuilder().withPhone("91234567").build()));
```
Comment (AlphaPyke) [inline src/test/java/seedu/address/model/person/PersonContainsKeywordsPredicateTest.java:101]: should there be testing for partial match with birthday?

File: src/test/java/seedu/address/model/person/PersonContainsKeywordsPredicateTest.java [hunk 3]
```diff
@@ -16,115 +14,141 @@ public class PersonContainsKeywordsPredicateTest {
...
+        // Search by email (no match)
+        searchCriteria = Map.of("email", "bob@example.com");
+        predicate = new PersonContainsKeywordsPredicate(searchCriteria);
+        assertFalse(predicate.test(new PersonBuilder().withEmail("alice@example.com").build()));
+
+        // Search by tag (no match)
+        searchCriteria = Map.of("tag", "colleague");
+        predicate = new PersonContainsKeywordsPredicate(searchCriteria);
+        assertFalse(predicate.test(new PersonBuilder().withTags("friend").build()));
+    }
```
Comment (AlphaPyke) [inline src/test/java/seedu/address/model/person/PersonContainsKeywordsPredicateTest.java:130]: should there be search by birthday here as well?

**Comment by meatyturtle** [comment]
LGTM

**Pull request merged**